Persist onion addresses with "make dev-tor"

deeplow opened the pull request:
Makes the SecureDrop Workstation dev workflow easier by removing the need to setup standalone server VMs if one wants a persistent server address and authentication token.

The solution involves creating a docker volume. This will persist as long as the user does not manually remove it or moves to another system without porting the volume. Since this a dev environment, long term persistence should not be a concern, so this solution sufices.

Fixes #7115

## Status

Ready for review 

## Description of Changes

Fixes #7115.

Changes proposed in this pull request:

## Testing
Run `make dev-tor` multiple times and ensure that the same journalist config is generated.

## Deployment

Any special considerations for deployment? no

## Checklist

### If you made changes to the server application code:

- [ ] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you added or removed a file deployed with the application:

- [ ] I have updated AppArmor rules to include the change

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

Choose one of the following:

- [ ] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes, or will do so later
- [ ] I would appreciate help with the documentation
- [ ] These changes do not require documentation

### If you added or updated a reference to a production code dependency:

Production code dependencies are defined in:

- `admin/requirements.in`
- `admin/requirements-ansible.in`
- `securedrop/requirements/python3/requirements.in`
- `securedrop/requirements/python3/translation.in` (used in the build
  container)

If you changed another `requirements.in` file that applies only to development
or testing environments, then no diff review is required, and you can skip
(remove) this section.

Choose one of the following:

- [ ] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
- [ ] I would like someone else to do the diff review
- [ ] I am silencing an alert related to a production dependency, because (please explain below):

Commit:
Persist onion addresses with "make dev-tor"

Makes the SecureDrop Workstation dev workflow easier by removing
the need to setup standalone server VMs if one wants a persistent
server address and authentication token.

The solution involves creating a docker volume. This will persist
as long as the user does not manually remove it or moves to another
system without porting the volume. Since this a dev environment,
long term persistence should not be a concern, so this solution
sufices.

Fixes #7115

zenmonkeykstop commented:
This looks like a great addition @deeplow, with only a couple of things to address:
- lint shellcheck fails need correcting
- the `make clean` target should be updated to remove the container volume with `/var/lib/tor` in it
- Ctrl-C no longer seems to work to kill the services? unclear if that's as a result of this PR tho.

deeplow commented:
>    lint shellcheck fails need correcting
Done in 6efc371

>    the make clean target should be updated to remove the container volume with /var/lib/tor in it
Addressed in c1b8c31

>    Ctrl-C no longer seems to work to kill the services? unclear if that's as a result of this PR tho.

Addressed in 2e1f3e11. I wasn't keeping the `DOCKER_RUN_ARGUMENTS` and thus loosing the `-it` params.

zenmonkeykstop commented on devops/clean:
```diff
@@ -36,6 +36,11 @@ function remove_unwanted_files() {
         build/*.deb
 
+    # Remove any Onion URL from make dev-tor
+    if docker volume sd-onion-services; then
```
Should this be `docker volume inspect sd-onion-services`? This is incorrect syntax (at least on my docker) so it will never return 0, and the volume will never be nuked.

deeplow replied:
You're right. This should be addressed in 890020f, where I also suppressed the result to `/dev/null`.

zenmonkeykstop reviewed (changes requested):
One problem with the cleanup script, looking good so far otherwise.

deeplow commented:
Rebased and squased. This is ready to merge, pending the resolution of the discussion.
